fe/fe-core/src/main/java/com/starrocks/qe/scheduler/slot/SlotSelectionStrategyV2.java

```diff
@@ -154,10 +160,29 @@ private void updateOptionsPeriodically() {
         QueryQueueOptions newOpts = QueryQueueOptions.createFromEnv();
         if (!newOpts.equals(opts)) {
             opts = newOpts;
-            requiringQueue = new WeightedRoundRobinQueue(opts.v2().getTotalSlots(), opts.v2().getNumWorkers());
+
+            // Change the schedule algorithm, move pending slots to new queue
+            SlotScheduleAlgorithm newQueue = null;
+            if (opts.getPolicy().equals(QueryQueueOptions.SchedulePolicy.SWRR)) {
+                newQueue = new WeightedRoundRobinQueue(opts.v2().getTotalSlots(), opts.v2().getNumWorkers());
+            } else if (opts.getPolicy().equals(QueryQueueOptions.SchedulePolicy.SJF)) {
+                newQueue = new ShortJobFirstAlgo();
+            } else {
+                Preconditions.checkState(false, "unknown schedule policy: " + opts.getPolicy());
+            }
+            if (requiringQueue != null) {
+                while (requiringQueue.isEmpty()) {
+                    newQueue.add(requiringQueue.poll());
+                }
```

Comment:
Maybe while non-empty?
```suggestion
                while (!requiringQueue.isEmpty()) {
                    newQueue.add(requiringQueue.poll());
                }
```